pkg/upgraders/healthcheck_pdb.go

```diff
@@ -105,3 +106,38 @@ func checkDvoMetrics(c client.Client, dvo dvo.DvoClientBuilder, logger logr.Logg
 
 	return "", nil
 }
+
+// validateMaxUnavailable function will return false for failures if
+// MaxUnavailable for a PDB is 0 or 0% if defined
+func validateMaxUnavailable(p policyv1.PodDisruptionBudget, l logr.Logger) (bool, error) {
+	if p.Spec.MaxUnavailable != nil {
+		switch p.Spec.MaxUnavailable.Type {
+		case intstr.Int:
+			if p.Spec.MaxUnavailable.IntVal == 0 {
+				l.Info(fmt.Sprintf("MaxUnavailable 0 found in PodDisruptionBudget: %s/%s\n", p.Namespace, p.Name))
+				return false, fmt.Errorf("found a PodDisruptionBudget with MaxUnavailable set to 0")
+			}
+		case intstr.String:
+			if p.Spec.MaxUnavailable.StrVal == "0%" {
+				l.Info(fmt.Sprintf("MaxUnavailable 0%% found in PodDisruptionBudget: %s/%s\n", p.Namespace, p.Name))
```

Comment:
Should be single % in the error log?

Reply:
Hey Tony! Well, I wanted to show in logs that it's a percentage and not an integer so wanted to print % as a literal character. We need to have double %% to present it as a single % in logs. 

Refer https://pkg.go.dev/fmt:
> %%	a literal percent sign; consumes no value

Reply:
Example log:
```
4-09-19T05:52:56.564725387Z level=info logger=controller_upgradeconfig msg="MinAvailable 100% found in PodDisruptionBudget: ns1/ns1-pdb\n" Request.Namespace=openshift-managed-upgrade-operator Request.Name=managed-upgrade-config
```